lib/schema/generateSchema.js

```diff
@@ -85,32 +90,12 @@ exports.generateSchema = function generateSchema(next) {
 				}
 				else {
 					self.logger.debug('schema file updates are available, regenerating');
-					// TODO: Don't throw out the whole schema file, just update the changed objects...
-					self._cachedSchema = null;
-					self.cleanGeneratedModels();
-					self.describeGlobal(confDir, hashedFilename, next);
+					cleanModelsAndGenerateNewSchema();
 				}
 			});
 	}
-
-	function useCachedVersion() {
-		self.logger.debug('reading the cached schema file', cachedFileName);
-		fs.readFile(cachedFileName, 'UTF-8', function readFile(err, contents) {
-			if (!next) {
-				return;
-			}
-			if (err) {
-				next(self.parseError(err));
-			}
-			else {
-				next(null, JSON.parse(contents));
-			}
-		});
-
-	}
-
-	// refresh the schema on a schedule (if schemaRefresh is set)
-	if (self.config.schemaRefresh) {
+    // refresh the schema on a schedule (if schemaRefresh is set)
+	if (self.config.schemaRefresh) { 
```

Comment:
We already have conditional logic for doing schema refresh or not. We can skip this check, shape the code bellow as a function and invoke it in the proper place.